lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb

```diff
@@ -254,16 +256,24 @@ def self.bodies_and_line_numbers(obj, aem)
     end
     private_class_method :bodies_and_line_numbers
 
-    def self.embedded_methods(workspace, method_obj)
-      method_obj.embedded_methods.collect do |name|
+    def self.embedded_methods(workspace, method_obj, current_items, top)
+      method_obj.embedded_methods.each do |name|
         method_name, klass, ns = embedded_method_name(name)
         match_ns = workspace.overlay_method(ns, klass, method_name)
         cls = ::MiqAeClass.find_by_fqname("#{match_ns}/#{klass}")
         aem = ::MiqAeMethod.find_by(:class_id => cls.id, :name => method_name) if cls
         raise MiqAeException::MethodNotFound, "Embedded method #{name} not found" unless aem
-        fqname = "#{match_ns}/#{klass}/#{method_name}"
-        $miq_ae_logger.info("Loading embedded method #{fqname}")
-        {:data => aem.data, :fqname => fqname}
+        fqname = "/#{match_ns}/#{klass}/#{method_name}"
+        if top == fqname
+          $miq_ae_logger.info("Skipping #{fqname}, cannot reference the top method")
```

Comment:
Also don't understand the "top" distinction

Reply:
@Fryguy thats to prevent circular referencing, a method can embed itself and we dont have any checks in the UI from stopping you doing that.

Reply:
Ah ok...sounds like then there's other ways to get circulars as well... a includes b and b includes a for example